include/pythontarget.h

```diff
@@ -184,6 +184,53 @@ typedef struct {
 } generic_port_capsule_struct;
 
 
+/**
+ * Special version of the template_input_output_port_struct
+ * for dynamic arrays.
+ * FIXME: This is currently kept for compatibility reasons
+ * and should be removed soon.
+ **/
+typedef struct {
+    PyObject_HEAD
+    PyObject* value;
+    bool is_present;
+    int num_destinations;
+    lf_token_t* token;
+    int length;
+} generic_port_instance_with_token_struct;
+
+/**
+ * Python wrapper for the tag_t struct in the C target.
+ **/
+typedef struct {
+    PyObject_HEAD
+    tag_t tag;
```

Comment:
I think that since there are separate get_time and get_microstep getters for Tag, it's better to also individually store them in the type itself. You can prepend the `time` and `microstep` variables with a `_` or `__` to signify that they are for internal use, but maybe that's not entirely necessary since changing these values doesn't affect the execution of the runtime.

Reply:
As an example, a user might want to just define a `Tag t` in their program and then subsequently manipulate `t.time` and `t.microstep` as they wish. I think allowing this is harmless.

Reply:
The user can still use t.time and t.microstep to retrieve the values (via ```.tp_getset```), which I implemented getter functions and omitted setter functions (so the attributes are readonly).

Reply:
The reason I made the attributes readonly is to prevent the user from doing something like: ```a = Tag(0, 0); del a.time```

Reply:
Okay, I see your reason. It makes sense.